### DISCUSSION TITLE
Prisma integration testing with Jest

### DISCUSSION THREAD

**Discussion by engelkes-finstreet** [prisma/prisma#12204] [Q&A] [answered]
Hey 👋 
I just got started with Jest and Prisma testing and followed the Prisma guide for [integration testing](https://www.prisma.io/docs/guides/testing/integration-testing) 
One test is working just fine. But I get a problem after adding a second test

Let's assume that `createPlayer(numberOfPlayers: number)` is a function that creates the desired number of players. 

```
beforeEach(async () => {
  return createPlayer(1)
})

afterEach(async () => {
  return clearDatabase()  
})

//unrelated test
it("should test something else", () => {})

```

```
beforeEach(async () => {
  return createPlayer(2)
})

afterEach(() => {
  return clearDatabase()
})

it("should get the correct number of players", async () => {
  const numberOfPlayers = await getNumberOfPlayers()
  expect(numberOfPlayers).toBe(2)
})
```

In my first test case I need to create a player to link it to another model. In my second test I want to create two players to get the correct number of players. Running all tests fails, I guess because the tests run in parallel and the `beforeEach` function of both tests is executed at the same time --> thus the `numberOfPlayers` is `3` in my second test and not `2` as I expect it to be. 

Is there a way to fix this? Have I done something wrong?

**Comment by engelkes-finstreet** [comment, marked as answer]
With some help from the `Reactiflux Discord` I figured out to seth `maxWorkers: 1` in my `jest.config.js` which solves the problem!
Is there a way to achieve the parallel test runs with my setup or should I just go with serial order here?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
